### ISSUE TITLE
Create custom responsive form component

### ISSUE THREAD

**Issue by knjk04** [knjk04/book-project#259]
We are currently using a form inside a dialog in a few different places: adding a book, adding a custom shelf and setting a reading goal. It would be nice if this was wrapped around a component that is responsive (currently it isn't). 

This would fix the problem in #57

**Comment by knjk04** [comment]
No longer needed after #282
